staking/types/messages.go

```diff
@@ -18,16 +22,189 @@ type StakingMessage interface {
+	msg.ValidatingPubKey = msgCreateValJSON.ValidatingPubKey
+	msg.Amount = msgCreateValJSON.Amount
+
+	return nil
+}
+
+// ValidateBasic quick validity check
+func (msg MsgCreateValidator) ValidateBasic() error {
+	// note that unmarshaling from bech32 ensures either empty or valid
+	if msg.StakingAddress.Big().Uint64() == 0 {
+		return errors.New("[CreateValidator] address is empty")
+	}
+	emptyKey := shard.BlsPublicKey{}
+	if bytes.Compare(msg.ValidatingPubKey[:], emptyKey[:]) == 0 {
+		return errors.New("[CreateValidator] invalid BLS public key")
+	}
+	if msg.Description == (Description{}) {
+		return errors.New("[CreateValidator] description must be included")
+	}
+	if msg.Commission == (CommissionRates{}) {
+		return errors.New("[CreateValidator] commission must be included")
+	}
+	if msg.Amount.Cmp(msg.MinSelfDelegation) > 0 {
+		return errors.New("[CreateValidator] stake amount must be >= MinSelfDelegation")
+	}
+
+	return nil
+}
+
+// MsgEditValidator - struct for editing a validator
+type MsgEditValidator struct {
+	Description
+	StakingAddress common.Address `json:"staking_address" yaml:"staking_address"`
+
+	CommissionRate    Dec      `json:"commission_rate" yaml:"commission_rate"`
+	MinSelfDelegation *big.Int `json:"min_self_delegation" yaml:"min_self_delegation"`
+}
+
+// MsgEditValidatorJSON - struct for editing a validator for JSON
+type MsgEditValidatorJSON struct {
+	Description
+	StakingAddress common.Address `json:"staking_address" yaml:"staking_address"`
+
+	// TODO: allow update of bls public key
+	CommissionRate    Dec      `json:"commission_rate" yaml:"commission_rate"`
+	MinSelfDelegation *big.Int `json:"min_self_delegation" yaml:"min_self_delegation"`
+}
+
+// NewMsgEditValidator creates a new MsgEditValidator.
+func NewMsgEditValidator(
+	description Description, stakingAddress common.Address, commissionRate Dec, minSelfDelegation *big.Int) MsgEditValidator {
+
+	return MsgEditValidator{
+		Description:       description,
+		StakingAddress:    stakingAddress,
+		CommissionRate:    commissionRate,
+		MinSelfDelegation: minSelfDelegation,
+	}
+}
+
+// Type ...
+func (msg MsgEditValidator) Type() string { return "edit_validator" }
+
+// Signer ...
+func (msg MsgEditValidator) Signer() common.Address { return msg.StakingAddress }
+
+// MsgDelegate - struct for bonding transactions
+type MsgDelegate struct {
+	DelegatorAddress common.Address `json:"delegator_address" yaml:"delegator_address"`
+	ValidatorAddress common.Address `json:"validator_address" yaml:"validator_address"`
+	Amount           *big.Int       `json:"amount" yaml:"amount"`
+}
+
+// NewMsgDelegate creates a new MsgDelegate.
+func NewMsgDelegate(
+	validatorAddress common.Address, delegatorAddress common.Address, amount *big.Int) MsgDelegate {
+
+	return MsgDelegate{
+		DelegatorAddress: delegatorAddress,
+		ValidatorAddress: validatorAddress,
+		Amount:           amount,
+	}
+}
+
+// Type ...
+func (msg MsgDelegate) Type() string { return "delegate" }
```

Comment:
const string should be const, no need to make fresh string

Reply:
Done.